core/src/modules/ops/import.ts

```diff
@@ -66,11 +68,22 @@ export namespace ImportOps {
   }
 
   export async function associateProfile(_import: Import) {
-    const {
-      profile,
-      isNew,
-      // will throw if there are no unique profile properties
-    } = await ProfileOps.findOrCreateByUniqueProfileProperties(_import.data);
+    let run: Run;
+    let schedule: Schedule;
+    let source: Source;
+
+    run = await Run.findOne({ where: { id: _import.creatorId } });
+    if (run && run.creatorType === "schedule")
+      schedule = await Schedule.findOne({ where: { id: run.creatorId } });
+    if (schedule)
+      source = await Source.findOne({ where: { id: schedule.sourceId } });
+
+    // will throw if there are no unique profile properties
+    const { profile, isNew } =
+      await ProfileOps.findOrCreateByUniqueProfileProperties(
+        _import.data,
+        source
```

Comment:
`source` will only be set when imports are created via a schedule's run, which will trigger verification inside `findOrCreateByUniqueProfileProperties`. Other non-schedule-created imports will still be able to create profiles at will, is that right?

Schedules should be the only type of run that creates profiles, so I think this is 👍🏼

Reply:
It's the other way around actually - if you don't provide a source (eg: the run is crated by an internal run or something like that) you cannot create a new profile if you can't find one.  So this means that only Schedules can create properties now.

Some of the updates to the test suite reflect this.

Reply:
Oh- that would make more sense, but it doesn't seem like that's what's happening.

I'm seeing the `prevents import when unable to create profile from secondary source` test for `associateProfile`, which tests imports created by a schedule from a different source don't create profiles. 

However, I'm not seeing any test for imports not created from a schedule (e.g. internal runs or group runs). From what I'm reading in the code, it seems like if no schedule is passed, `canCreateNewProfile` is set to `true`, so the profile will be created:

```js
const canCreateNewProfile = source
  ? (await Property.findAllWithCache())
      .filter((p) => p.unique === true && p.sourceId === source.id)
      .map((p) => p.key)
      .filter((key) => !!hash[key]).length > 0
  : true; # <- if source is falsy, canCreateNewProfile = true
```

maybe that final `true` should be `false`?

Reply:
Whoops! I was wrong in my reply, and your first 👍 note was correct.  That's what the code and test say - that Profile creation by methods that don't have a source (eg: internal run) is still allowed.  I'm going to dig into why that is...

Reply:
Ok! https://github.com/grouparoo/grouparoo/pull/2192/commits/765d9708721745e83164408f7295a8d16eb169de should make things make a lot more sense!